javascript/ql/lib/semmle/javascript/Expr.qll

```diff
@@ -2823,9 +2823,9 @@ class DynamicImportExpr extends @dynamic_import, Expr, Import {
   /**
    * Gets the second "argument" to the import expression, that is, the `Y` in `import(X, Y)`.
    *
-   * For example, gets the `{ assert: { type: "json" }}` expression in the following:
+   * For example, gets the `{ with: { type: "json" }}` expression in the following:
    * ```js
-   * import('foo', { assert: { type: "json" }})
+   * import('foo', { with: { type: "json" }})
    * ```
    */
   Expr getImportAttributes() { result = this.getChildExpr(1) }
```

Comment:
There's an annoying discrepancy here, in that this doesn't return the same type of thing as for `ImportDeclaration` and `ExportDeclaration`:

```js
// getImportAttributes() = { type: "json" }
import "foo" with { type: "json" }

// getImportAttributes() = { with: { type: "json" } }
import("foo", { with: { type: "json" }} )
```

The enclosing `with` property is omitted in the first version and (by necessity) included in the second version.

That's why this was already called `getImportAttributes` and other one was called `getImportAssertion`, to clarify that they don't do the same thing. The term "import attribute" was initially something I came up with to avoid the ambiguity. Unfortunately, the new proposal _also_ uses the term "import attribute", except with a different meaning. It seems in the new proposal, this value is [called the "options bag"](https://github.com/tc39/proposal-import-attributes#dynamic-import), although I'd suggest a slightly nicer name `getImportOptions()`.

TLDR: Please rename this to `getImportOptions()` and deprecate the old name (as it does not in fact return the import attributes).

Reply:
Thanks for spotting that, I hadn't realised that they were actually different. I just assumed you somehow had a crystal ball to foretell the future name of the feature ;-)

Reply:
Can't we somehow make `getImportAttributes()` peel off the `with:` and return `{ type: "json" }` for 
```
import("foo", { with: { type: "json" }} )
```

Although I guess this gets tricky for cases like
```
let options = { with: { type: "json" }}
import("foo", options)
```

Reply:
> Can't we somehow make `getImportAttributes()` peel off the `with:` and return `{ type: "json" }` for

I don't think this part of the standard library should do that. I think that should be up the query writers (and they can use API-graphs).

Reply:
Agree with @erik-krogh, and in any case we still need `getImportOptions()` for the general case. Also let's not worry too much about building convenience predicates that we might never need.